packages/mui-icons-material/scripts/create-typings.mjs

```diff
@@ -29,16 +29,23 @@ type SvgIconComponent = typeof SvgIcon;
 ${files.map((file) => `export const ${normalizeFileName(file)}: SvgIconComponent;`).join('\n')}
 `;
 
-  return fse.writeFile(path.resolve(TARGET_DIR, 'index.d.ts'), contents, 'utf8');
+  await fs.writeFile(path.resolve(targetDir, 'index.d.ts'), contents, 'utf8');
 }
 
 // Generate TypeScript.
 async function run() {
-  await fse.ensureDir(TARGET_DIR);
+  await Promise.all([
+    fs.mkdir(TARGET_DIR, { recursive: true }),
+    fs.mkdir(TARGET_DIR_ESM, { recursive: true }),
+  ]);
   console.log(`\u{1f52c}  Searching for modules inside "${chalk.dim(SRC_DIR)}".`);
   const files = await glob('!(index)*.js', { cwd: SRC_DIR });
-  const typings = files.map((file) => createIconTyping(file));
-  await Promise.all([...typings, createIndexTyping(files)]);
+  await Promise.all([
+    createIconTyping(TARGET_DIR),
+    createIndexTyping(TARGET_DIR, files),
+    createIconTyping(TARGET_DIR_ESM),
+    createIndexTyping(TARGET_DIR_ESM, files),
+  ]);
```

Comment:
Given we're sharing one type file for all icons now, isn't this script unnecessary? We can manually create this 4 files once

Reply:
We can potentially create the icon typing manually, but the index must be generated. I think we're treating the lib folder as generated.